packit_service/worker/handlers/mixin.py

```diff
@@ -80,6 +82,86 @@ def koji_build_helper(self) -> KojiBuildJobHelper:
         return self._koji_build_helper
 
 
+class GetKojiBuildData(Protocol):
+    @property
+    @abstractmethod
+    def nvr(self) -> str:
+        ...
+
+    @property
+    @abstractmethod
+    def build_id(self) -> int:
+        ...
+
+    @property
+    @abstractmethod
+    def dist_git_branch(self) -> str:
+        ...
+
+    @property
+    @abstractmethod
+    def state(self) -> KojiBuildState:
+        ...
+
+
+class GetKojiBuildDataFromKojiBuildEventMixin(GetKojiBuildData, GetKojiBuildEvent):
+    @property
+    def nvr(self) -> str:
+        return self.koji_build_event.nvr
+
+    @property
+    def build_id(self) -> int:
+        return self.koji_build_event.build_id
+
+    @property
+    def dist_git_branch(self) -> str:
+        return self.koji_build_event.git_ref
+
+    @property
+    def state(self) -> KojiBuildState:
+        return self.koji_build_event.state
+
+
+class GetKojiBuildDataFromKojiServiceMixin(ConfigMixin, GetKojiBuildData):
+    _build: Optional[Any] = None
+    _koji_helper: Optional[KojiHelper] = None
+
+    @property
+    def koji_helper(self):
+        if not self._koji_helper:
+            self._koji_helper = KojiHelper()
+        return self._koji_helper
+
+    @property
+    def build(self):
+        if not self._build:
+            self._build = self.koji_helper.get_latest_build_in_tag(
+                package=self.project.repo,
+                tag=self.dist_git_branch,
+            )
+            if not self._build:
+                raise PackitException(
+                    f"No build found for package={self.project.repo} and tag={self.dist_git_branch}"
+                )
+        return self._build
+
+    @property
+    def nvr(self) -> str:
+        return self.build["nvr"]
+
+    @property
+    def build_id(self) -> int:
+        return self.build["build_id"]
```

Comment:
could you please maybe add a link to the documentation where can one find what the build dictionary looks like?